src/util/StringUtils.h

```diff
@@ -83,26 +84,14 @@ inline string normalizeSpaces(const string& orig);
 inline string escapeForJson(const string& orig);
 
 //! Strips any sequence of characters in s from the left and right of the text.
-inline string strip(const string& text, const string& s);
-
-inline string strip(const string& text, const char* s);
-
-//! Strips any sequence of c from the left and right of the text.
-inline string strip(const string& text, char c);
-
-//! Strips any sequence of c from the left of the text.
-inline string lstrip(const string& text, char c);
-
-inline string lstrip(const string& text, string s);
-
-inline string lstrip(const string& text, const char* s);
-
-//! Strips any sequence of c from the tight of the text.
-inline string rstrip(const string& text, char c);
-
-inline string rstrip(const string& text, string s);
-
-inline string rstrip(const string& text, const char* s);
+template <class T>
+inline string strip(const string& text, const T& s);
```

Comment:
I think it would be more explicit to comment the constraints of the template parameters like "T must be a single char or a string(_view) like type." (not a good example but I think you get what I mean).